backends/ubpf/ubpfRegister.cpp

```diff
@@ -58,6 +58,29 @@ namespace UBPF {
         }
     }
 
+    void UBPFRegister::emitInstance(EBPF::CodeBuilder *builder) {
+        cstring keyTypeStr;
+        if (keyType != nullptr && keyType->is<IR::Type_Bits>()) {
+            auto tb = keyType->to<IR::Type_Bits>();
+            auto scalar = new UBPFScalarType(tb);
+            keyTypeStr = scalar->getAsString();
+        } else {
+            keyTypeStr = cstring("struct ") + keyTypeName.c_str();;
```

Comment:
Is this always true? Maybe you want to code defensively and add a BUG_CHECK for the type?

Reply:
I've refactored the `emitInstance()` method and added some additional BUG_CHECKs.